doc/classes/ProjectSettings.xml

```diff
@@ -553,7 +553,7 @@
 			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a static inferred type uses a [Variant] as initial value, which makes the static type to also be Variant.
 		</member>
 		<member name="debug/gdscript/warnings/inferred_declaration" type="int" setter="" getter="" default="0">
-			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a variable, constant, or parameter has an implicitly inferred static type.
+			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a variable, constant, or parameter has an implicitly inferred static type. In GDScript, type inference is performed by declaring a variable with [code]:=[/code] instead of [code]=[/code] and leaving out the type specifier. For example, [code]var x := 1[/code] will [i]infer[/i] the [int] type, while [code]var x: int = 1[/code] explicitly declares the variable as [int].
```

Comment:
In addition to variables, type inference is also available for constants and function parameters (yes, they are also variables, but sometimes they are singled out as a separate category). But this is probably too much detail.

Reply:
That makes sense, but I think we can leave this as-is since it's almost exclusively used on variables in practice. (There was also some discussion on removing type inference for function parameters in 4.0, but it didn't happen in the end.)